src/main/java/com/datatorrent/netlet/OptimizedEventLoop.java

```diff
@@ -59,7 +59,8 @@ public boolean add(SelectionKey key)
       if (key == null) {
         return false;
       }
-      if (pos >= keys.length) {
+      // An extra slot is needed to store the null delimiter when keys are retrieved
+      if (pos >= (keys.length - 1)) {
```

Comment:
Good catch. To avoid subtraction can you move check and array copy after pos is incremented. What do you think of changing condition from >= to == and using pos instead of key.length in arraycopy?